### PR TITLE
Added types for 'ipcheck'

### PR DESCRIPTION
**Pull request by bengry**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by andy-ms** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

File: types/ipcheck/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6"],
+        "noImplicitAny": false,
```
Comment (andy-ms) [inline types/ipcheck/tsconfig.json]: This should be `true` for new packages, same for `strictNullChecks`

File: types/ipcheck/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (andy-ms) [inline types/ipcheck/tslint.json]: Please use a file containing just `{ "extends": "tslint.json" }`. Let me know if you have trouble with any particular lint rule.

File: types/ipcheck/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+interface IPCheckConstructor {
+    new (input: string): IPCheck;
+}
+
+interface IPCheckStatic extends IPCheckConstructor {
+    match(ip: IPCheck | string, cidr: IPCheck | string): boolean;
+}
+
+declare module "ipcheck" {
```
Comment (andy-ms) [inline types/ipcheck/index.d.ts]: Prefer to write this file like:
```ts
declare const ipcheck: ipcheck.IPCheckStatic;
export = ipcheck;
export as namespace ipcheck;
namespace ipcheck {
    ....interfaces...
}
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/ipcheck/tslint.json (hunk 2 above)
Comment (bengry) [reply types/ipcheck/tslint.json]: Just the 'interface-name` rule. All interfaces here start with an `IP`, causing the rule of:
```
interface-name  interface name must not have an "I" prefix
```
to trigger needlessly. I disabled this now with a:
```typescript
// tslint:disable-next-line:interface-name
```
but I don't like it very much. If you have any better suggestions, I'm all ears.

Comment (andy-ms) [reply types/ipcheck/tslint.json]: Just disable that rule for now then.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**